docs/USAGE.md

```diff
@@ -138,6 +138,17 @@ You should see a bunch of logs that describe the steps of the test, from:
 * Starting the containers (total of 50 as 50 is the default number of nodes for this test)
 * You will see the logs that describe each node connecting to the others and executing a kademlia find-peers action.
 
+
+In order to run the tests with the local:exec runner, there are a few things that must be taken care of first.
+1. install required test software
+  * redis (redis.io)
+  * prometheus (prometheus.io)
```

Comment:
Are `prometheus` and `prometheus-pushgateway` *required* components of `testground` now? I thought that monitoring components (specially when ran with local:exec, local:docker, etc.) are mostly optional.

If `prometheus` crashes for some reason, what is the expected outcome of a testplan (if we assume that the testplan is correct and passing) ?

Reply:
I think that is a completely valid concern. I'm not particularly a fan of having these as required infrastructure, but having them opportunistically available would probably be a good thing.

What do you think about this? -- if the pushgateway is unavailable and can't start it, don't fail the healthcheck but don't give a healthy OK message either.

This way it starts automatically if it can, and doesn't prevent you from running without it.

Reply:
I regard them as required. We offer a metric API that wires directly onto Prometheus. Plans using that API will just end up sending stuff to /dev/null if those components are optional, and the environment doesn’t have them. Why would there be any benefit in making them optional?

Reply:
In general you don't always care about metrics. For example in tests. Would we have to set up prometheus as part of our unit and integration tests for modules that are instrumented?

Most previous projects I've work on have always had a way to disable metrics, so that they don't incur performance cost, when you don't need them - generally this is done for tracing, but there is no reason to not do it for metrics as well.